fixture/model.cpp

```diff
@@ -0,0 +1,485 @@
+#include "model.h"
```

Comment:
I stripped all of the logic out of the framer, removed any notion of having to deal with an ISD and we have a model that is just an interface. Logic can then get added, as needed, to do things like throw appropriate warnings, etc. See groundToImage for an example of this.